gpytorch/kernels/kernel.py

```diff
@@ -172,7 +172,7 @@ def __init__(
                 if not isinstance(lengthscale_prior, Prior):
                     raise TypeError("Expected gpytorch.priors.Prior but got " + type(lengthscale_prior).__name__)
                 self.register_prior(
-                    "lengthscale_prior", lengthscale_prior, lambda m: m.lengthscale, lambda m, v: m._set_lengthscale(v)
+                    "lengthscale_prior", lengthscale_prior, self._lengthscale_param, self._lengthscale_closure
```

Comment:
I guess this should also work here?
```suggestion
                    "lengthscale_prior", lengthscale_prior, "lengthscale", self._lengthscale_closure
```

Reply:
I don't think this works (making this change causes 31 tests to fail). Note that `_lengthscale_param` returns `m.lengthscale` and not `self.lengthscale`.

Reply:
Hmm not sure I understand why it doesn’t, seems like the closure definition here should do the same? https://github.com/cornellius-gp/gpytorch/blob/32cde571f8ee0bf8d1a523b23e51cce57aaa5045/gpytorch/module.py#L235

Either way, we can figure that out later (or I’m missing sth obvious). Accepting  this as is.

Reply:
Sounds good! Looks like merging is blocked though until the docs can be built successfully. Looking at https://readthedocs.org/projects/gpytorch/builds/15684237/ it looks like the issue is a warning that is unrelated to this PR.